From 4950186e4da5c8547478cd634c35d69ada6d8037 Mon Sep 17 00:00:00 2001
From: Summer Qin <summer.qin@arm.com>
Date: Fri, 9 Jul 2021 17:35:22 +0800
Subject: [PATCH 1/3] Fix wrong expected return value in test 46 and 47

Update the expected return value in test 46 and 47
according to the PSA Crypto spec 1.0.

Signed-off-by: Summer Qin <summer.qin@arm.com>
---
 api-tests/dev_apis/crypto/test_c046/test_data.h | 4 ++--
 api-tests/dev_apis/crypto/test_c047/test_data.h | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/api-tests/dev_apis/crypto/test_c046/test_data.h b/api-tests/dev_apis/crypto/test_c046/test_data.h
index bfe6553..39b2597 100644
--- a/api-tests/dev_apis/crypto/test_c046/test_data.h
+++ b/api-tests/dev_apis/crypto/test_c046/test_data.h
@@ -54,7 +54,7 @@ static const test_data check1[] = {
  0x9d, 0xf3, 0x3f, 0x47, 0xb4, 0xb1, 0x16, 0x99, 0x12, 0xba, 0x4f, 0x53, 0x68,
  0x4b, 0x22}, 64, 28,
  PSA_KEY_USAGE_SIGN_HASH, PSA_ALG_CMAC,
- PSA_ERROR_NOT_SUPPORTED
+ PSA_ERROR_INVALID_ARGUMENT
 },
 
 {"Test psa_mac_compute - Invalid usage\n", PSA_KEY_TYPE_HMAC,
@@ -65,7 +65,7 @@ static const test_data check1[] = {
  0x9d, 0xf3, 0x3f, 0x47, 0xb4, 0xb1, 0x16, 0x99, 0x12, 0xba, 0x4f, 0x53, 0x68,
  0x4b, 0x22}, 64, 28,
  PSA_KEY_USAGE_EXPORT, PSA_ALG_HMAC(PSA_ALG_SHA_224),
- PSA_ERROR_NOT_SUPPORTED
+ PSA_ERROR_NOT_PERMITTED
 },
 
 {"Test psa_mac_compute - truncated MAC too small\n", PSA_KEY_TYPE_HMAC,
diff --git a/api-tests/dev_apis/crypto/test_c047/test_data.h b/api-tests/dev_apis/crypto/test_c047/test_data.h
index 3c250dc..54f6a28 100644
--- a/api-tests/dev_apis/crypto/test_c047/test_data.h
+++ b/api-tests/dev_apis/crypto/test_c047/test_data.h
@@ -66,7 +66,7 @@ static const test_data check1[] = {
  0x0b, 0xf1, 0x2b, 0x88, 0x1d, 0xc2, 0x00, 0xc9, 0x83, 0x3d, 0xa7, 0x26, 0xe9,
  0x37, 0x6c, 0x2e, 0x32, 0xcf, 0xf7}, 32,
  PSA_KEY_USAGE_VERIFY_HASH, PSA_ALG_CMAC,
- PSA_ERROR_NOT_SUPPORTED
+ PSA_ERROR_INVALID_ARGUMENT
 },
 
 {"Test psa_mac_verify - Invalid usage\n", PSA_KEY_TYPE_HMAC,
@@ -77,7 +77,7 @@ static const test_data check1[] = {
  0x0b, 0xf1, 0x2b, 0x88, 0x1d, 0xc2, 0x00, 0xc9, 0x83, 0x3d, 0xa7, 0x26, 0xe9,
  0x37, 0x6c, 0x2e, 0x32, 0xcf, 0xf7}, 32,
  PSA_KEY_USAGE_EXPORT, PSA_ALG_HMAC(PSA_ALG_SHA_256),
- PSA_ERROR_NOT_SUPPORTED
+ PSA_ERROR_NOT_PERMITTED
 },
 
 {"Test psa_mac_verify - Truncated MAC too large\n", PSA_KEY_TYPE_HMAC,
-- 
2.17.1

